orcoursetrion/tests/test_github.py

```diff
@@ -113,15 +119,119 @@ def test_unknown_errors(self):
+        git_hub.add_team_repo(self.ORG, self.TEST_REPO, self.TEST_TEAM)
+
+    @httpretty.activate
+    def test_lib_add_team_repo_no_teams(self):
+        """Test what happens when we don't have any teams
+        """
+
+        # Register for team list API
+        httpretty.register_uri(
+            httpretty.GET,
+            '{url}orgs/{org}/teams'.format(
+                url=self.URL,
+                org=self.ORG,
+            ),
+            body=partial(self.callback_team_list, status_code=404)
+        )
+        git_hub = GitHub(self.URL, self.OAUTH2_TOKEN)
+        # See how we handle no teams
+        with self.assertRaisesRegexp(
+                GitHubUnknownError,
+                'No teams found in {0} organization'.format(self.ORG)
+        ):
+            git_hub.add_team_repo(self.ORG, self.TEST_REPO, self.TEST_TEAM)
+
+    @httpretty.activate
+    def test_lib_add_team_repo_team_not_found(self):
+        """Test what happens when the team isn't in the org
+        """
+
+        # Register for team list API
+        httpretty.register_uri(
+            httpretty.GET,
+            '{url}orgs/{org}/teams'.format(
+                url=self.URL,
+                org=self.ORG,
+            ),
+            body=partial(self.callback_team_list)
+        )
+        git_hub = GitHub(self.URL, self.OAUTH2_TOKEN)
+        with self.assertRaises(GitHubNoTeamFound):
+            git_hub.add_team_repo(self.ORG, self.TEST_REPO, 'foobar')
+
+    @httpretty.activate
+    @httpretty.activate
+    def test_lib_add_team_repo_fail(self):
+        """Test what happens when the repo can't be added to the team
+        """
+
+        # Register for team list API
+        httpretty.register_uri(
+            httpretty.GET,
+            '{url}orgs/{org}/teams'.format(
+                url=self.URL,
+                org=self.ORG,
+            ),
+            body=partial(self.callback_team_list)
+        )
+        # Register for team repo add API
+        httpretty.register_uri(
+            httpretty.PUT,
+            '{url}teams/{id}/repos/{org}/{repo}'.format(
+                url=self.URL,
+                id=self.TEST_TEAM_ID,
+                org=self.ORG,
+                repo=self.TEST_REPO
+            ),
+            body=partial(self.callback_team_repo, status_code=422)
+        )
+
+        git_hub = GitHub(self.URL, self.OAUTH2_TOKEN)
+        with self.assertRaisesRegexp(GitHubUnknownError, json.dumps({
+                "message": "Validation Failed",
+        })):
+            git_hub.add_team_repo(self.ORG, self.TEST_REPO, self.TEST_TEAM)
+
     @mock.patch('orcoursetrion.actions.github.config')
     @httpretty.activate
     def test_actions_create_export_repo_success(self, config):
         """Test the API call comes through as expected.
         """
         config.ORC_GH_OAUTH2_TOKEN = self.OAUTH2_TOKEN
-        config.ORC_GH_API_URL = self.URL
+        config.ORC_GH_API_URL = self.URL[:-1]
```

Comment:
nit: Is this for cutting off the slash at the end? Had to think about it for a moment, but it might be good to have a note here.